### PR TITLE
flcl: add new versions

### PR DESCRIPTION
**Pull request by rbberger**

### CODE REVIEW

**Review by womeld — APPROVED** [review]
@spackbot summoned me and I think the changes look a-ok.

**Review by womeld — DISMISSED** [review]
Changes look great.

File: var/spack/repos/builtin/packages/flcl/package.py [hunk 1]
```diff
@@ -11,18 +11,23 @@ class Flcl(CMakePackage):
...
-    url      = "https://github.com/kokkos/kokkos-fortran-interop/releases/download/0.3/0.3.tar.gz"
+    url      = "https://github.com/kokkos/kokkos-fortran-interop/releases/download/0.5.0/flcl-0.5.0.tar.gz"
 
     maintainers = ['womeld', 'agaspar']
 
     version('develop', branch='develop')
     version('0.3', sha256='0586b981b976588d8059e5bf1bf71fb5a7153ea950c7e2b562a3d812fefee56e')
+    version('0.4.0', sha256='0fe327906a991262866b126a7d58098eb48297148f117fd59a2dbcc14e76f394')
+    version('0.5.0', sha256='bfd9b9092904eab1135d3bb4c458a50653b3325c176a722af56f158da0a16f19')
+    version('0.99.0', sha256='edb8310154e5e5cf315dad63cd59f13b2537e0ba698869ce9757b04e38047464')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/flcl/package.py]: The Spack convention is for versions to be sorted high-to-low (i.e. 0.99.0 first, 0.5.0 second, etcetera).

File: var/spack/repos/builtin/packages/flcl/package.py [hunk 2]
```diff
@@ -11,18 +11,23 @@ class Flcl(CMakePackage):
...
 
     depends_on('kokkos')
-    depends_on('cmake@3.17:', type='build')
+    depends_on('cmake@3.19:', type='build')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/flcl/package.py]: Does this dependency apply to `0.3` as well, or from a specific newer version on?

If the latter, you could have something like:

```
    depends_on('cmake@3.17:', type='build', when='@:0.3')
    depends_on('cmake@3:19:', type='build', when='@0.4.0:')
```

File: var/spack/repos/builtin/packages/flcl/package.py [hunk 3]
```diff
@@ -11,18 +11,23 @@ class Flcl(CMakePackage):
 
     homepage = "https://github.com/kokkos/kokkos-fortran-interop"
     git      = "https://github.com/kokkos/kokkos-fortran-interop.git"
-    url      = "https://github.com/kokkos/kokkos-fortran-interop/releases/download/0.3/0.3.tar.gz"
+    url      = "https://github.com/kokkos/kokkos-fortran-interop/releases/download/0.5.0/flcl-0.5.0.tar.gz"
 
     maintainers = ['womeld', 'agaspar']
 
     version('develop', branch='develop')
     version('0.3', sha256='0586b981b976588d8059e5bf1bf71fb5a7153ea950c7e2b562a3d812fefee56e')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/flcl/package.py]: The URL change changes the checksum for this version to the following:

```suggestion
    version('0.3', sha256='fc18c8fa3ae33db61203b647ad9025d894612b0faaf7fe07426aaa8bbfa9e703')
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/flcl/package.py (hunk 3 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/flcl/package.py]: Confirmed the URLs for the new versions are correct.

File: var/spack/repos/builtin/packages/flcl/package.py (hunk 2 above)
Comment (womeld) [reply var/spack/repos/builtin/packages/flcl/package.py]: Looking back through the older versions of FLCL, it appears that the dependency on Cmake 3.19 was introduced in FLCL 0.5.0. So if this block was changed to e.g. 
```
depends_on('cmake@3.17:', type='build', when='@:0.3')
depends_on('cmake@3:19:', type='build', when='@0.5.0:')
```
I think that would work, if I understand Spack logic correctly.

File: var/spack/repos/builtin/packages/flcl/package.py (hunk 1 above)
Comment (womeld) [reply var/spack/repos/builtin/packages/flcl/package.py]: This makes sense to me.

Comment (rbberger) [reply var/spack/repos/builtin/packages/flcl/package.py]: Changed the order.

File: var/spack/repos/builtin/packages/flcl/package.py (hunk 2 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/flcl/package.py]: Changed the dependency to use `cmake@3.17:` for <= 0.4.0 and `cmake@3.19:` for >= 0.5.0

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**